rpc/handlers_test.go

```diff
@@ -1341,3 +1343,132 @@ func TestClassAt(t *testing.T) {
 		assertEqualCairo0Class(t, cairo0Class, class)
 	})
 }
+
+func TestEvents(t *testing.T) {
+	testDB := pebble.NewMemTest()
+	t.Cleanup(func() {
+		require.NoError(t, testDB.Close())
+	})
+	chain := blockchain.New(testDB, utils.GOERLI2, utils.NewNopZapLogger())
+
+	client, closeFn := feeder.NewTestClient(utils.GOERLI2)
+	t.Cleanup(closeFn)
+	gw := adaptfeeder.New(client)
+
+	for i := 0; i < 7; i++ {
+		b, err := gw.BlockByNumber(context.Background(), uint64(i))
+		require.NoError(t, err)
+		s, err := gw.StateUpdate(context.Background(), uint64(i))
+		require.NoError(t, err)
+		require.NoError(t, chain.Store(b, s, nil))
+	}
+
+	handler := rpc.New(chain, nil, utils.MAINNET, utils.NewNopZapLogger())
+	from := utils.HexToFelt(t, "0x49d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7")
+	args := &rpc.EventsArg{
+		EventFilter: &rpc.EventFilter{
+			FromBlock: &rpc.BlockID{Number: 0},
+			ToBlock:   &rpc.BlockID{Latest: true},
+			Address:   from,
+			Keys:      []*felt.Felt{},
+		},
+		ResultPageRequest: &rpc.ResultPageRequest{
+			ChunkSize:         100,
+			ContinuationToken: "",
+		},
+	}
+
+	t.Run("filter non-existent", func(t *testing.T) {
+		t.Run("block number", func(t *testing.T) {
+			args.ToBlock = &rpc.BlockID{Number: 55}
+			_, err := handler.Events(args)
+			require.Equal(t, rpc.ErrBlockNotFound, err)
```

Comment:
What is the reason for not checking the result of `handler.Events(args)`? At the very least you should check the len of the filtered events which are returned, right?

Reply:
It is returning an error, so no events will be returned.